metadoc-js/src/main/scala/metadoc/MetadocApp.scala

```diff
@@ -54,31 +50,46 @@ object MetadocApp extends js.JSApp {
     )
     monaco.languages.Languages.registerDefinitionProvider(
       ScalaLanguageExtensionPoint.id,
-      new ScalaDefinitionProvider(attrs, index)
+      new ScalaDefinitionProvider(index)
     )
     monaco.languages.Languages.registerReferenceProvider(
       ScalaLanguageExtensionPoint.id,
-      new ScalaReferenceProvider(attrs, index)
+      new ScalaReferenceProvider(index)
     )
     monaco.languages.Languages.registerDocumentSymbolProvider(
       ScalaLanguageExtensionPoint.id,
-      new ScalaDocumentSymbolProvider(attrs, index)
+      new ScalaDocumentSymbolProvider(index)
     )
-    dom.document.getElementById("title").textContent = fileName
 
     val options = jsObject[IEditorConstructionOptions]
     options.readOnly = true
+    val overrides = jsObject[IEditorOverrideServices]
+    val editorService = new MetadocEditorService
+    overrides.textModelResolverService = MetadocTextModelService
+    overrides.editorService = editorService
+    val editor = monaco.editor.Editor.create(app, options, overrides)
+    editor.asInstanceOf[js.Dynamic].getControl = { () =>
+      // NOTE: getControl() is defined on SimpleEditor and is called when changing files.
+      editor
+    }
+    editorService.editor = editor
+    editor.onDidChangeModel((arg1: IModelChangedEvent) => {
+      val path = arg1.newModelUrl.path
+      dom.document.getElementById("title").textContent = path
+      dom.window.location.hash = "#/" + path
+    })
 
-    val editor = monaco.editor.Editor.create(app, options)
-
-    val model = monaco.editor.Editor.createModel(
-      value = contents,
-      language = "scala",
-      uri = monaco.Uri.parse(s"file:$fileName")
-    )
-    editor.setModel(model)
-
+    dom.window.onhashchange = { e: Event =>
```

Comment:
Yup! Line numbers would definitely be nice, I opened https://github.com/scalameta/metadoc/issues/38